client/components/select-dropdown/test/index.js

```diff
@@ -1,253 +1,215 @@
-		} );
-	} );
-
-	/**
-	 * Utilities
-	 */
-
-	function mountDropdown( attach = false ) {
-		const dropdownOptions = getDropdownOptions();
-		return mount(
-			<SelectDropdown options={ dropdownOptions } />,
-			attach ? { attachTo: document.body } : undefined
-		);
-	}
-
-	function shallowRenderDropdown( props ) {
-		const dropdownOptions = getDropdownOptions();
-		return shallow( <SelectDropdown options={ dropdownOptions } { ...props } /> );
-	}
-
-	function getDropdownOptions() {
-		return [
-			{ value: 'status-options', label: 'Statuses', isLabel: true },
-			{ value: 'published', label: 'Published' },
-			{ value: 'scheduled', label: 'Scheduled' },
-			{ value: 'drafts', label: 'Drafts' },
-			null,
-			{ value: 'trashed', label: 'Trashed' },
-		];
-	}
-
-	function createKeyEvent( keyCode ) {
-		return {
-			keyCode,
-			preventDefault: jest.fn(),
-		};
-	}
+function getDropdownOptions() {
+	return [
+		{ value: 'status-options', label: 'Statuses', isLabel: true },
+		{ value: 'published', label: 'Published' },
+		{ value: 'scheduled', label: 'Scheduled' },
+		{ value: 'drafts', label: 'Drafts' },
+		null,
+		{ value: 'trashed', label: 'Trashed' },
+	];
+}
+
+function renderDropdown( props ) {
+	return render( <SelectDropdown options={ getDropdownOptions() } { ...props } /> );
+}
+
+describe( 'component rendering', () => {
+	test( 'should render a list with the provided options', () => {
+		renderDropdown();
+		expect( screen.getByRole( 'presentation' ) ).toHaveTextContent( 'Statuses' );
+		expect( screen.queryAllByRole( 'listitem' ) ).toHaveLength( 4 );
+	} );
+
+	test( 'should render a separator in place of any falsy option', () => {
+		renderDropdown();
+		expect( screen.queryAllByRole( 'separator' ) ).toHaveLength( 1 );
+	} );
+
+	test( 'should be initially closed', () => {
+		renderDropdown();
+		expect( screen.getByRole( 'button' ) ).toHaveAttribute( 'aria-expanded', 'false' );
+	} );
+
+	test( 'should execute toggleDropdown when clicked', async () => {
+		renderDropdown();
+
+		const btn = screen.getByRole( 'button' );
+		expect( btn ).toHaveAttribute( 'aria-expanded', 'false' );
+
+		await userEvent.click( btn );
+		expect( btn ).toHaveAttribute( 'aria-expanded', 'true' );
+
+		await userEvent.click( btn );
+		expect( btn ).toHaveAttribute( 'aria-expanded', 'false' );
+	} );
+
+	test( 'should not respond when clicked when disabled', async () => {
+		renderDropdown( { disabled: true } );
+
+		const btn = screen.getByRole( 'button' );
+		expect( btn ).toHaveAttribute( 'aria-disabled', 'true' );
+
+		await userEvent.click( btn );
+		expect( btn ).toHaveAttribute( 'aria-expanded', 'false' );
+	} );
+
+	test( 'should be possible to open the dropdown via keyboard', async () => {
+		renderDropdown();
+
+		const btn = screen.getByRole( 'button' );
+
+		await userEvent.tab();
+		expect( btn ).toHaveFocus();
+
+		await userEvent.keyboard( '[Space]' );
+		expect( btn ).toHaveAttribute( 'aria-expanded', 'true' );
```

Comment:
Pretty cool test! I like the example of how to use keyboard navigation with RTL tests :)